library/src/text/shaping.rs

```diff
@@ -75,17 +75,34 @@ impl ShapedGlyph {
     /// in a line until the line fills the available width. However, some
     /// scripts (notably Chinese and Japanese) don't use spaces.
     ///
-    /// In Japanese typography, the convention is to insert space evenly
-    /// between all glyphs. I assume it's the same in Chinese.
+    /// In Japanese/Chinese typography, the convention is to insert space evenly
+    /// between all glyphs.
     pub fn is_justifiable(&self) -> bool {
-        self.is_space() || is_spaceless(self.c.script())
+        self.is_space() || self.is_cjk() || self.is_cjk_punctuaction()
+    }
+
+    pub fn is_cjk(&self) -> bool {
+        use Script::*;
+        matches!(self.c.script(), Hiragana | Katakana | Han)
     }
-}
 
-/// Does this script separate its words using spaces?
-fn is_spaceless(script: Script) -> bool {
-    use Script::*;
-    matches!(script, Hiragana | Katakana | Han)
+    pub fn is_cjk_punctuaction(&self) -> bool {
+        matches!(self.c, '，' | '。' | '、' | '：' | '；')
+    }
+
+    /// The stretchability and shrinkability of the character.
+    pub fn stretch(&self) -> (Em, Em) {
```

Comment:
i think this should be split up into two function for stretchability and shrinkability.

Reply:
Fixed in 92d4b382722c0a2f8bb5db0e65ab25997cf290d8